addons/payment_authorize/models/payment.py

```diff
@@ -173,7 +173,7 @@ def authorize_s2s_form_validate(self, data):
             if len(cc_expiry) != 2 or any(not i.isdigit() for i in cc_expiry):
                 return False
             try:
-                if datetime.now().strftime('%y%m') > datetime.strptime('/'.join(cc_expiry), '%m/%y').strftime('%y%m'):
+                if datetime.now().strftime('%y%m') > datetime(int(cc_expiry[1]), int(cc_expiry[0]),1).strftime('%y%m'):
```

Comment:
this will crash when the user fill the form manually as 2 digit year will be used in that case, I guess the only solution is a manual fallback
```
try:
    if datetime.now().strftime('%y%m') > datetime.strptime('/'.join(cc_expiry), '%m/%y').strftime('%y%m'):
        return False
except ValueError:
    try:
        if datetime.now().strftime('%y%m') > datetime.strptime('/'.join(cc_expiry), '%m/%Y').strftime('%y%m'):
            return False
    except ValueError:
       return False
```

Reply:
I just checked, datetime() can manage years with 4 or 2 digits... 
Is it ok or I apply your solution?

Reply:
:man_facepalming: I was testing with ipython2.7
It should be ok, I had no other remarks
@nim-odoo r+?

Reply:
since it is v11, I'd make it compatible with Python 2.7 :)

Reply:
really? I thought the last odoo shipped on 2.7 was 10

Reply:
Yes, but when released Odoo 11.0 did support (unofficially) Python 2.7. So if we know that it can crash in Python 2.7, better avoid it. By the way, the minimum required version for 11.0 is Python 3.5. I don't know how this behaves in 3.5.

Reply:
@nim-odoo So I guess I apply @agr-odoo suggested fix, right?
If so, do I also apply the same fix on v12+?

Reply:
yep